analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,233 @@
+                #    TCHE = ptcs_IP_signif_sorted[TCHE_index]
+                #    TCHE_x = ptcs_x_at_time_0[ptcs_IP_signif.index(TCHE)]
+                #    TCHE_y = ptcs_y_at_time_0[ptcs_IP_signif.index(TCHE)]
+                #if TCHE_x**2 + TCHE_y**2 - (0.05)**2 > 0:
+                #    TCHE = -99
+                #    TCHP = -99
+                #    TCHE_IP = -99
+                #    TCHP_IP = -99
+                #    TCHE_x = -99
+                #    TCHE_y = -99
+                #   TCHE_z = -99
+                #else:
+                #    TCHE_IP = ptcs_IP[ptcs_IP_signif.index(TCHE)]
+                #    TCHE_z = ptcs_z_at_time_0[ptcs_IP_signif.index(TCHE)]
+                #    TCHP = ptcs_IP_signif_sorted[TCHE_index + 1]
+                #    TCHP_IP = ptcs_IP[ptcs_IP_signif.index(TCHP)]
+            #if TCHE > 30 and event.K0s == 0 and event.Kp == 0 and event.L0 == 0 and event.S0 == 0 and event.Sp == 0 and event.Sm == 0:
+            #    import pdb; pdb.set_trace()
+            
+            jet.tags['IP_b_LL'] = IP_b_LL  if self.tag_IP_b_LL  else None
+            jet.tags['IPs_b_LL']= IPs_b_LL if self.tag_IPs_b_LL else None
+            jet.tags['TCHE'] = TCHE
+            jet.tags['TCHP'] = TCHP
+            jet.tags['TCHE_IP'] = TCHE_IP
+            jet.tags['TCHP_IP'] = TCHP_IP
+            jet.tags['TCHE_x'] = TCHE_x
+            jet.tags['TCHE_y'] = TCHE_y
+            jet.tags['TCHE_z'] = TCHE_z
+            jet.tags['TCHE_xy'] = (TCHE_x**2+TCHE_y**2)**0.5
+            jet.tags['TCHE_pt'] = TCHE_pt
+            jet.tags['TCHE_dr'] = TCHE_dr
+            
+            if hasattr(event, 'K0s') == True :
+                jet.tags['K0s'] = event.K0s
+            else :
+                jet.tags['K0s'] = -99
+            if hasattr(event, 'Kp') == True :
+                jet.tags['Kp'] = event.Kp
+            else :
+                jet.tags['Kp'] = -99
+            if hasattr(event, 'L0') == True :
+                jet.tags['L0'] = event.L0
+            else :
+                jet.tags['L0'] = -99
+            if hasattr(event, 'S0') == True :
+                jet.tags['S0'] = event.S0
+            else :
+                jet.tags['S0'] = -99
+            if hasattr(event, 'Sp') == True :
+                jet.tags['Sp'] = event.Sp
+            else :
+                jet.tags['Sp'] = -99
+            if hasattr(event, 'Sm') == True :
+                jet.tags['Sm'] = event.Sm
+            else :
+                jet.tags['Sm'] = -99
+            if hasattr(event, 'Muons') == True :
+                jet.tags['Muons'] = event.Muons
+            else :
+                jet.tags['Muons'] = -99
+        
         
+    def beginLoop(self, setup):
+        super(ImpactParameter, self).beginLoop(setup)
+        if hasattr(self.cfg_ana, 'num_IP') == False :
+            self.tag_IP_b_LL = False
+        else :
+            if hasattr(self.cfg_ana, 'denom_IP') == False :
+                self.tag_IP_b_LL = False
```

Comment:
Le 30 juin 2016 à 10:26, lucastorterotot <notifications@github.com> a écrit :

> In analyzers/ImpactParameter.py:
> 
> >          
> > +    def beginLoop(self, setup):
> > +        super(ImpactParameter, self).beginLoop(setup)
> > +        if hasattr(self.cfg_ana, 'num_IP') == False :
> > +            self.tag_IP_b_LL = False
> > +        else :
> > +            if hasattr(self.cfg_ana, 'denom_IP') == False :
> > +                self.tag_IP_b_LL = False
> Well, I do not understand what happened, in my file there is an exception, I wrote
> if hasattr(self.cfg_ana, 'denom_IP') == False :
> self.tag_IP_b_LL = False
> 
> raise AttributeError('You specified a numerator without a denominator for the log likelihood based on IP')
> 
> —
> 

Could be that the code has not been committed to your repository and is only there as a local file change in the working directory. 
Or maybe the commit has not been pushed to your master branch on github. 



> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>